### ISSUE TITLE
Training and testing of the model

### ISSUE THREAD

**Issue by kxie2022** [cliffren/PENCIL#8]
Hello, I noticed that in the fit_transform and transform functions in pencil.py the same "data" and "labels" variables are used for both the training and testing of the model--testing and training datasets should not be the same? If an external testing dataset is allowed, how can I specify the external dataset? Thank you.

**Comment by cliffren** [comment]
The training and test datasets can be different, we actually implemented fit and transform separately, you can call fit on the training dataset and transform on the test dataset exactly as you are used to with the sklearn library, but you need to pay attention to the standardization step for the test dataset in it, like the general machine learning workflow.

![image](https://github.com/cliffren/PENCIL/assets/32982291/270382fa-ae47-4e73-8cbc-39f8cb8e79d1)
